central/globaldb/v2backuprestore/backup/generators/dbs/postgres.go

```diff
@@ -30,102 +23,41 @@ func NewPostgresBackup(db *pgxpool.Pool) *PostgresBackup {
 	}
 }
 
-// PostgresBackup is an implementation of a postgres connection pool
+// PostgresBackup is an implementation of a StreamGenerator which writes a backup of PostgresDB to the input io.Writer.
 type PostgresBackup struct {
 	db *pgxpool.Pool
 }
 
-// getPostgresSize Method to calculate size
-func (bu *PostgresBackup) getPostgresSize(ctx context.Context) (int64, error) {
-	row := bu.db.QueryRow(ctx, "SELECT pg_database_size('postgres')")
-	var count int64
-	if err := row.Scan(&count); err != nil {
-		return 0, err
-	}
-	return count, nil
-}
-
-// WriteDirectory writes a backup of RocksDB to the input path.
-func (bu *PostgresBackup) WriteDirectory(ctx context.Context) (string, error) {
-	centralConfig := config.GetConfig()
-	password, err := os.ReadFile(dbPasswordFile)
-	if err != nil {
-		log.Fatalf("pgsql: could not load password file %q: %v", dbPasswordFile, err)
-		return "", err
-	}
-	source := fmt.Sprintf("%s password=%s", centralConfig.CentralDB.Source, password)
-	sourceMap := bu.parseSource(source)
-
-	config, err := pgxpool.ParseConfig(source)
+// WriteTo writes a backup of Postgres to the writer
+func (bu *PostgresBackup) WriteTo(ctx context.Context, out io.Writer) error {
+	sourceMap, config, err := globaldb.GetPostgresConfig()
 	if err != nil {
 		log.Fatalf("Could not parse postgres config: %v", err)
-		return "", err
-	}
-
-	backupPath, err := bu.findScratchPath(ctx)
-	if err != nil {
-		return "", err
+		return err
 	}
 
 	// Set the options for pg_dump from the connection config
 	options := []string{
-		"-U",
-		config.ConnConfig.User,
-		"-h",
-		config.ConnConfig.Host,
-		"-p",
-		strconv.FormatUint(uint64(config.ConnConfig.Port), 10),
 		"-d",
 		config.ConnConfig.Database,
-		"-Fd",
-		"-f",
-		backupPath,
-		"-j",
-		"5",
+		"-Fc", // Custom format, compressed hopefully supports stdin to restore
```

Comment:
We cannot easily check the data from database dump. Can you create a jira to create/find a tool to convert db dump to human readable format.

Reply:
Looks like that is as simple as running pg_restore without a database name.

https://stackoverflow.com/questions/21429335/how-do-i-convert-a-binary-pgdump-compressed-to-a-plain-sql-file

That dumps the SQL, I'm not terribly sure that we will get much more readable than that.  So I'm not sure a jira ticket is necessary.